File: src/main/java/net/nunnerycode/bukkit/mythicdrops/api/socketting/SocketEffect.java
package net.nunnerycode.bukkit.mythicdrops.api.socketting;

import org.bukkit.entity.LivingEntity;

public interface SocketEffect {

    int getIntensity();

    int getDuration();

    EffectTarget getEffectTarget();

    int getRadius();

    boolean isAffectsWielder();

    boolean isAffectsTarget();

    void apply(LivingEntity target);

}

File: src/main/kotlin/com/tealcube/minecraft/bukkit/mythicdrops/api/socketting/SocketEffect.kt
package com.tealcube.minecraft.bukkit.mythicdrops.api.socketting

import org.bukkit.entity.LivingEntity

interface SocketEffect {
    val intensity: Int
    val duration: Int
    val effectTarget: EffectTarget
    val radius: Int
    val chanceToTrigger: Double
    val affectsWielder: Boolean
    val affectsTarget: Boolean

    fun apply(target: LivingEntity?)
}